src/lib/theme.ts

```diff
@@ -0,0 +1,15 @@
+const themes = ['light', 'dark'] as const;
+export type ThemeName = (typeof themes)[number];
+
+export const setTheme = (theme: ThemeName) => {
+	localStorage.setItem('theme', theme);
+	if (theme === 'light') document.documentElement.classList.add('light');
+	else document.documentElement.classList.remove('light');
+};
+
+export const getTheme = () => {
+	let theme = localStorage.getItem('theme');
+	if (!theme || !themes.includes(theme as ThemeName))
+		theme = matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light';
+	return theme as ThemeName;
```

Comment:
:question: Is there a way to avoid the initial flicker / re render when light mode?

Reply:
Looks like probably not unless we can resolve this server side using a cookie instead, but i'd prefer keeping it client side.